app/assets/js/rtm-upload-terms.js

```diff
@@ -1,139 +1,166 @@
-					if ( undefined !== args && false !== args && undefined !== args.src && 'activity' === args.src ) {
-						rtp_display_terms_warning( form.find( '.rtmedia-upload-terms' ), rtmedia_upload_terms_data.message );
-					} else {
-						rtp_display_terms_warning( terms_conditions_checkbox.parent( '.rtmedia-upload-terms' ), rtmedia_upload_terms_data.message );
-					}
+                }
+
+                if ( ! terms_conditions_checkbox.is( ':checked' ) ) {
+                    whats_new_submit.removeAttr( 'disabled' );
+                    whats_new_submit.removeClass( 'loading' );
+                    if ( undefined !== args && false !== args && undefined !== args.src && 'activity' === args.src ) {
+                        rtp_display_terms_warning( form.find('.rtmedia-upload-terms'), rtmedia_upload_terms_data.message );
+                    } else {
+                        rtp_display_terms_warning( terms_conditions_checkbox.parent( '.rtmedia-upload-terms' ), rtmedia_upload_terms_data.message );
+                    }
+					
 					return false;
-				}
-			}
+                }
+            }
+			
 			return true;
+        }
+    );
+
+    /**
+     * When Select Attribute for media [ rtmedia-custom-attributes: Add-Ons ] Issue:8,
+     * This Hook returns false that's why the attributes were not saved,
+     * Added: 'return true;'
+     * By: Yahil
+     */
+    rtMediaHook.register( 'rtmedia_js_after_file_upload', function () {
+
+        var terms_conditions_checkbox = jQuery( '#rtmedia-upload-container #rtmedia_upload_terms_conditions' );
+        if ( 1 === terms_conditions_checkbox.length ) {
+            terms_conditions_checkbox.removeAttr( 'checked' );
+            jQuery( '.rt_alert_msg' ).remove();
 		}
-	);
+		
+        return true;
+    });
 
 	/**
-	 * When Select Attribute for media [ rtmedia-custom-attributes: Add-Ons ] Issue:8,
-	 * This Hook returns false that's why the attributes were not saved,
-	 * Added: 'return true;'
-	 * By: Yahil
+	 * Uncheck the terms checkbox after the activity is posted successfully.
 	 */
-	rtMediaHook.register( 'rtmedia_js_after_file_upload', function () {
-
-		var terms_conditions_checkbox = jQuery( '#rtmedia-upload-container #rtmedia_upload_terms_conditions' );
-		if ( terms_conditions_checkbox.length > 0 ) {
-			terms_conditions_checkbox.removeAttr( 'checked' );
-			jQuery( '.rt_alert_msg' ).remove();
-		}
-		return true;
-	});
-
-	rtMediaHook.register( 'rtmedia_js_after_activity_added', function() {
-		var rtmedia_terms_conditions = $( '#rtmedia_upload_terms_conditions' );
-		if ( rtmedia_terms_conditions && rtmedia_terms_conditions.is(':checked') ) {
-			rtmedia_terms_conditions.prop( 'checked', false );
-		}
-	} );
+    rtMediaHook.register( 'rtmedia_js_after_activity_added', function () {
+        var rtmedia_terms_conditions = $( '#rtmedia_upload_terms_conditions' );
+        if ( rtmedia_terms_conditions && rtmedia_terms_conditions.is( ':checked' ) ) {
```

Comment:
No need to check for `rtmedia_terms_conditions`. jQuery won't return `undefined`. You can instead check for the `length`.